DQM/SiPixelHeterogeneous/plugins/SiPixelCompareRecHits.cc

```diff
@@ -0,0 +1,283 @@
+  hBsizeXDiff_ = iBook.book1D("rechitsizeXDiffBpix","SizeX difference of rechits in BPix; rechit sizex difference (Reference - Target)", 21, -10.5, 10.5);
+  hFsizeXDiff_ = iBook.book1D("rechitsizeXDiffFpix","SizeX difference of rechits in FPix; rechit sizex difference (Reference - Target)", 21, -10.5, 10.5);
+  hBsizeYDiff_ = iBook.book1D("rechitsizeYDiffBpix","SizeY difference of rechits in BPix; rechit sizey difference (Reference - Target)", 21, -10.5, 10.5);
+  hFsizeYDiff_ = iBook.book1D("rechitsizeYDiffFpix","SizeY difference of rechits in FPix; rechit sizey difference (Reference - Target)", 21, -10.5, 10.5);
+  hBposXDiff_ = iBook.book1D("rechitsposXDiffBpix","x-position difference of rechits in BPix; rechit x-pos difference (Reference - Target)", 1000, -10, 10);
+  hFposXDiff_ = iBook.book1D("rechitsposXDiffFpix","x-position difference of rechits in FPix; rechit x-pos difference (Reference - Target)", 1000, -10, 10);
+  hBposYDiff_ = iBook.book1D("rechitsposYDiffBpix","y-position difference of rechits in BPix; rechit y-pos difference (Reference - Target)", 1000, -10, 10);
+  hFposYDiff_ = iBook.book1D("rechitsposYDiffFpix","y-position difference of rechits in FPix; rechit y-pos difference (Reference - Target)", 1000, -10, 10);
+}
+
+template<typename T>
+void SiPixelCompareRecHits<T>::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  // monitorpixelRecHitsSoAAlpaka
+  edm::ParameterSetDescription desc;
+  // The default case for the comparison is still CUDA vs CUDA; the {Reference|Target}Alpaka
+  // input tags are set with "placeholders"; the specific tags for the other comparisons are defined
+  // in DQM/SiPixelHeterogeneous/python/SiPixelHeterogeneousDQM_FirstStep_cff.py
+  desc.add<edm::InputTag>("pixelHitsReferenceCUDA", edm::InputTag("siPixelRecHitsPreSplittingSoA@cpu"));
+  desc.add<edm::InputTag>("pixelHitsTargetCUDA", edm::InputTag("siPixelRecHitsPreSplittingSoA@cuda"));
+  desc.add<edm::InputTag>("pixelHitsReferenceAlpaka", edm::InputTag("siPixelRecHitsPreSplittingSoA@cpu"));
+  desc.add<edm::InputTag>("pixelHitsTargetAlpaka", edm::InputTag("siPixelRecHitsPreSplittingSoA@cpu"));
+  desc.add<std::string>("topFolderName", "SiPixelHeterogeneous/PixelRecHitsCompareGPUvsCPU");
+  desc.add<double>("minD2cut", 0.0001);
+  desc.add<std::string>("case", "CUDA");
+  descriptions.addWithDefaultLabel(desc);
+}
+
+using SiPixelPhase1CompareRecHits = SiPixelCompareRecHits<pixelTopology::Phase1>;
+using SiPixelPhase2CompareRecHits = SiPixelCompareRecHits<pixelTopology::Phase2>;
+using SiPixelHIonPhase1CompareRecHits = SiPixelCompareRecHits<pixelTopology::HIonPhase1>;
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+DEFINE_FWK_MODULE(SiPixelPhase1CompareRecHits);
+DEFINE_FWK_MODULE(SiPixelPhase2CompareRecHits);
+DEFINE_FWK_MODULE(SiPixelHIonPhase1CompareRecHits);
```

Comment:
```suggestion
DEFINE_FWK_MODULE(SiPixelPhase1CompareRecHitsSoA);
DEFINE_FWK_MODULE(SiPixelPhase2CompareRecHitsSoA);
DEFINE_FWK_MODULE(SiPixelHIonPhase1CompareRecHitsSoA);
```

See [this comment](https://github.com/cms-sw/cmssw/pull/43964#issuecomment-1966013353).